rust/src/io/object_store.rs

```diff
@@ -283,11 +283,8 @@ impl ObjectStore {
 
         if !expanded_path.try_exists()? {
             std::fs::create_dir_all(expanded_path)?;
-        } else if !expanded_path.is_dir() {
-            return Err(Error::IO {
-                message: format!("{} is not a lance directory", str_path),
-            });
```

Comment:
The only times you use `ObjectStore` you then turn around and immediately access `inner`.  Would it be better to have some kind of `object_store::ObjectStore` factory function instead of going through the proxy/adapter/facade `ObjectStore`?

Reply:
Yeah something like that could work. I think the ObjectStore layer could use a refactor at some point. We augment object store for local filesystem, which we might want to keep, but we also inject dataset-specific stuff like commit hooks.